skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,596 @@
+import numpy as np
+from skimage import util
+cimport numpy as np
+cimport cython
+from libc.math cimport exp
+
+ctypedef np.float32_t DTYPE_t
+
+cdef eps = 1.e-8
+
+
+@cython.boundscheck(False)
+cdef inline float patch_distance_2d(DTYPE_t [:, :] p1,
+                                    DTYPE_t [:, :] p2,
+                                    DTYPE_t [:, ::] w, int s):
+    cdef int i, j
+    cdef int center = s / 2
+    # Check if central pixel is too different in the 2 patches
+    cdef float tmp_diff = p1[center, center] - p2[center, center]
+    cdef float init = w[center, center] * tmp_diff * tmp_diff
+    if init > 1:
+        return eps
+    cdef float distance = 0
+    for i in range(s):
+        # exp of large negative numbers will be 0, so we'd better stop
+        if distance > 5:
```

Comment:
I would suggest (1) replacing "5" with a module-scope "constant", `max_distance`, and (2) matching the value of said `max_distance` to `eps`, so that the breakpoint corresponds to minimum value of the distance. ie either increase `eps` to `exp(-5)` or decrease `max_distance` to `-log(1e-8)` (about 18). Does this make sense?